### PR TITLE
Retrieve DataUpload into result ConfigMap during volume snapshot restore.

### PR DESCRIPTION
**Pull request by blackpiglet**
Fix issue #6117.
Add CSI plugin needs builder functions.

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fix issue #6117.

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/restore/restore.go [hunk 1]
```diff
@@ -535,6 +535,20 @@ func (ctx *restoreContext) execute() (results.Result, results.Result) {
...
 
+	// Clean the DataUploadResult ConfigMaps
+	defer func() {
+		opts := []crclient.DeleteAllOfOption{
+			crclient.InNamespace(""),
+			crclient.MatchingLabels{
+				velerov1api.RestoreUIDLabel: string(ctx.restore.UID),
+			},
+		}
+		err := ctx.kbClient.DeleteAllOf(go_context.Background(), &v1.ConfigMap{}, opts...)
```
[the remark this answers is not in the document]
Comment (Lyndon-Li) [reply pkg/restore/restore.go:546]: OK, that should work.

**Review by Lyndon-Li — DISMISSED** [review]

**Review by Lyndon-Li — APPROVED** [review]

**Review by Lyndon-Li — APPROVED** [review]

**Comment by blackpiglet** [comment]
@sseago 
@shubham-pampattiwar 
Could you help to take a look?

File: pkg/apis/velero/v1/labels_annotations.go [hunk 2]
```diff
@@ -78,4 +78,27 @@ const (
...
+
+const (
+	AsyncOperationIDPrefixDataDownload AsyncOperationIDPrefix = "dd-"
+	AsyncOperationIDPrefixDataUpload   AsyncOperationIDPrefix = "du-"
+)
+
+type VeleroResourceUsage string
+
+const (
+	VeleroResourceUsageDataUploadResult VeleroResourceUsage = "DataUploadUsage"
```
Comment (shubham-pampattiwar) [inline pkg/apis/velero/v1/labels_annotations.go:103]: NIT: The label-key already has `usage` in it, maybe just keep the value as `DataUpload`

File: pkg/cmd/server/server.go [hunk 3]
```diff
@@ -551,6 +552,7 @@ var defaultRestorePriorities = restore.Priorities{
 		"volumesnapshotclass.snapshot.storage.k8s.io",
 		"volumesnapshotcontents.snapshot.storage.k8s.io",
 		"volumesnapshots.snapshot.storage.k8s.io",
+		"datauploads.velero.io",
```
Comment (shubham-pampattiwar) [inline pkg/cmd/server/server.go:555]: +1 @blackpiglet

**Review by shubham-pampattiwar — DISMISSED** [review]
One NIT, else LGTM! But I beleieve you need to rebase the PR due to conflicts.

File: pkg/apis/velero/v1/labels_annotations.go (hunk 2 above)
Comment (blackpiglet) [reply pkg/apis/velero/v1/labels_annotations.go]: Modified.

**Review by Lyndon-Li — APPROVED** [review]

**Review by qiuming-best — APPROVED** [review]

**Pull request merged**